### PR TITLE
Fix issues when source path is using Windows format

### PR DESCRIPTION
**Pull request by tgodzik**
Fixes https://github.com/scalameta/metals/issues/3262

### CODE REVIEW

**Review by adpi2 — APPROVED** [review]
LGTM!

How is it possible that those bugs are not catched by:
```

CI / windows-latest jdk-adopt@1.11 unit tests 1 / 2 (pull_request)
CI / windows-latest jdk-adopt@1.11 unit tests 2 / 2 (pull_request)
```

**Comment by tgodzik** [comment]
> LGTM!
> 
> How is it possible that those bugs are not catched by:
> 
> ```
> 
> CI / windows-latest jdk-adopt@1.11 unit tests 1 / 2 (pull_request)
> CI / windows-latest jdk-adopt@1.11 unit tests 2 / 2 (pull_request)
> ```

I think we operate on URIs and paths only in the tests :thinking: VS Code in Windows sends in the breakpoints as paths starting with `C:`

File: metals/src/main/scala/scala/meta/internal/metals/debug/SourcePathAdapter.scala [hunk 1]
```diff
@@ -44,7 +44,7 @@ private[debug] final class SourcePathAdapter(
     sourceUri.getScheme match {
       case "jar" =>
         Option(AbsolutePath(Paths.get(sourceUri)).toFileOnDisk(workspace))
-      case "file" => Some(sourcePath.toAbsolutePath)
+      case "file" => Some(AbsolutePath(Paths.get(sourceUri)))
```
Comment (dos65) [inline metals/src/main/scala/scala/meta/internal/metals/debug/SourcePathAdapter.scala:47]: Maybe extract `AbsolutePath(Paths.get(sourceUri))` into value as it's used in both cases?

Also, `toFileOnDisk`  might be used even for local files - in this case it won't create anything.

**Review by dos65 — APPROVED** [review]
LGTM

**Comment by adpi2** [comment]
> I think we operate on URIs and paths only in the tests thinking VS Code in Windows sends in the breakpoints as paths starting with C:

What about those 2 methods? They are supposed to adapt the path of the client so that Metals only deals with uri. So maybe it should have been called somewhere.

https://github.com/scalameta/metals/blob/db5f9e235607b9c80d387031d683f5b6a5968414/metals/src/main/scala/scala/meta/internal/metals/debug/ClientConfigurationAdapter.scala#L45-L59

**Comment by tgodzik** [comment]
> > I think we operate on URIs and paths only in the tests thinking VS Code in Windows sends in the breakpoints as paths starting with C:
> 
> What about those 2 methods? They are supposed to adapt the path of the client so that Metals only deals with uri. So maybe it should have been called somewhere.
> 

I think so, I will take a look next week to double and triple check everything and merge with the current status.

**Pull request merged**